src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlException.cs

```diff
@@ -99,6 +99,204 @@ public override void GetObjectData(SerializationInfo si, StreamingContext contex
+                        // (provider: Named Pipes Provider, error: 0 - Overlapped I/O operation is in progress)
+                        case 997:
+                        // SQL Error Code: 921
+                        // Database '%.*ls' has not been recovered yet. Wait and try again.
+                        case 921:
+                        // SQL Error Code: 669
+                        // The row object is inconsistent. Please rerun the query.
+                        case 669:
+                        // SQL Error Code: 617
+                        // Descriptor for object ID %ld in database ID %d not found in the hash table during attempt to un-hash it.
+                        // A work table is missing an entry. Rerun the query. If a cursor is involved, close and reopen the cursor.
+                        case 617:
+                        // SQL Error Code: 601
+                        // Could not continue scan with NOLOCK due to data movement.
+                        case 601:
+                        // SQL Error Code: 233
+                        // The client was unable to establish a connection because of an error during connection initialization process before login.
+                        // Possible causes include the following: the client tried to connect to an unsupported version of SQL Server;
+                        // the server was too busy to accept new connections; or there was a resource limitation (insufficient memory or maximum
+                        // allowed connections) on the server. (provider: TCP Provider, error: 0 - An existing connection was forcibly closed by
+                        // the remote host.)
+                        case 233:
+                        // SQL Error Code: 121
+                        // The semaphore timeout period has expired
+                        case 121:
+                        // SQL Error Code: 64
+                        // A connection was successfully established with the server, but then an error occurred during the login process.
+                        // (provider: TCP Provider, error: 0 - The specified network name is no longer available.)
+                        case 64:
+                        // DBNETLIB Error Code: 20
+                        // The instance of SQL Server you attempted to connect to does not support encryption.
+                        case 20:
+                            return true;
+                            // This exception can be thrown even if the operation completed successfully, so it's safer to let the application fail.
+                            // DBNETLIB Error Code: -2
+                            // Timeout expired. The timeout period elapsed prior to completion of the operation or the server is not responding. The statement has been terminated.
+                            //case -2:
+                    }
+                }
+
+                return false;
```

Comment:
You think it is not? In which cases there are multiple errors returned for an exception? Anyway, if we retry because there is at least one transient error, and that error disappear, we can perhaps expect that other non-transient error will disappear. If we fail early we will not be able to detect if they are related.

Reply:
> Anyway, if we retry because there is at least one transient error, and that error disappear, we can perhaps expect that other non-transient error will disappear. If we fail early we will not be able to detect if they are related.

I'm not very knowledgeable about SQL Server/SqlClient error handling (or which situation produces multiple errors), but generally speaking, the whole point of an error being non-transient is that it doesn't just go away when you retry the same operation. So I don't think it makes sense for IsTransient to return true when there's at least one non-transient error.

Reply:
I have reverted the check.

Reply:
The retry in the driver is based on whether any of the multiple errors is transient. So I believe it would be fair to return `IsTransient` based on the same condition check.